base/iterators.jl

```diff
@@ -873,13 +873,15 @@ julia> collect(Iterators.flatten((1:2, 8:9)))
 flatten(itr) = Flatten(itr)
 
 eltype(::Type{Flatten{I}}) where {I} = eltype(eltype(I))
+IteratorEltype(::Type{Union{}}) = EltypeUnknown()
```

Comment:
This can go in generator.jl along with the other standard methods for `IteratorEltype`; it's not specific to `flatten`.

Reply:
ok - I will move the line to `generator.jl`